### DISCUSSION TITLE
Load scripts in internal browser windows

### DISCUSSION THREAD

**Discussion by sdavidg** [xiaoxiaoflood/firefox-scripts#134] [General Support] [answered]
Is it possible to load scripts in internal browser windows? For example, in the debugger iframe:
![imagen](https://user-images.githubusercontent.com/4955944/168050472-8f5c9ac8-9a75-4552-a529-8770dd6863f9.png)

**Comment by xiaoxiaoflood** [comment]
userChromeJS works normally in every window and *frame* except Browser Toolbox. Because Browser Toolbox actually runs over a different profile, it's kinda independent. In order run things in Browser Toolbox, you need to copy your `chrome` folder into the profile used by Browser Toolbox, that is, `<your_profile_folder>/chrome_debugger_profile/`.

You can test it by installing StyloaiX and creating this style:

```css
* {
  color: red;
}
```

If you copy your `chrome` folder (including this style) to `chrome_debugger_profile` and restart Firefox, Browser Toolbox will be styled in red.
![image](https://user-images.githubusercontent.com/25161346/168083643-1ef08cba-6e36-4f05-86fb-44c163a37af7.png)

**Comment by xiaoxiaoflood** [comment, marked as answer]
If you're talking about *tab* DevTools, just use `// @include         about:devtools-toolbox` in the script. Example:


```js
// ==UserScript==
// @name            consoledev
// @author          xiaoxiaoflood
// @include         about:devtools-toolbox
// ==/UserScript==

console.log('hi');
```

Of course, `console.log` runs in browser context, not in tab context, so you'll only find the log in Browser Console (Ctrl+Shift+J).

**Reply by sdavidg** [reply]
This was what I was looking for, a way to modify certain elements of the debugger without the need to create an extension.

Thank you so much
